src/entities/trader/taker.ts

```diff
@@ -0,0 +1,64 @@
+import { ConnectError } from '@connectrpc/connect';
+import { rfqClient, parseQuoteResponse } from '../../utils';
+import type { ParsedQuoteResponse } from '../../utils';
+import type { QuoteRequest } from '../../lib/codegen/rfq_pb';
+import { Trader } from './base-trader';
+import type { TraderConstructorArgs } from './base-trader';
+
+export class Taker extends Trader {
+  public constructor(args: TraderConstructorArgs) {
+    super(args);
+  }
+
+  public async sendRfqRequests({
+    quoteRequest,
+    onQuoteResponse,
+    signal,
+  }: {
```

Comment:
Since `WebTaker` extends from `Taker`, it would be useful to abstract the method in such a way that the streaming is variant, but the business logic of handling the stream would be inherited from the base `Taker`.